src/shaders/stdosl.h

```diff
@@ -407,6 +407,35 @@ int hash (point p, float t) BUILTIN;
 
 
 // Displacement functions
+void displace (vector offset) {
+    P += offset;
+    N = normalize (calculatenormal (P));
+}
+void displace (float amp) {
+    displace (amp * N);
+}
+void displace (string space, float amp) {
+    float len = length (transform (space, N));
```

Comment:
Does this do the right thing if the scale is not uniform?

Reply:
I think. But it might depend on what you think is "right".

Reply:
:) Yeah...  I've been using this idiom forever. It makes intuitive sense, and it does give the user some form of non-uniform control over the displacement magnitude. However, lately I've switched to using the cube-root of abs(determinant(matrix(space,"common"))) as the scale factor. This change of heart is due to my experience using non-uniformly scaled coordinate systems to control displacement. With the simple length(transform(N)) method, the results are never quite what I expect or desire, and so the presumed control it allows doesn't, in my recent opinion, live up to the expectation. As a result, I've concluded that preserving a volumetric sense of relative scale is more useful in the rare instance when a user must use a non-uniformly scaled space to control displacement magnitude. Since displacement can be viewed as a volumetric operation, I now see this as making more intuitive sense. But, all that said, the simple approach works perfectly in the case of uniformly scaled coordinates, and is cheaper to compute. And since that's by far the most likely scenario, it may be the right implementation choice.